### DISCUSSION TITLE
V2G

### DISCUSSION THREAD

**Discussion by JohnMcLear** [dzurikmiroslav/esp32-evse#50] [Ideas]
Do you have any plan or roadmap to include V2H/V2G?  Thanks!

**Comment by dzurikmiroslav** [comment]
Hi,
This project is based on the SAE J1772 specification, it is basically analog communication.
V2G/V2H/V2L is described in ISO 15118, which uses digital power-line communication and uses the entire TCP-IP stack, so completely different hardware would be needed.

I haven't dealt with V2G yet, because:
The first problem is that ISO 15118 is not freely downloadable.
EV manufacturers are still not uniform in implementation.

So, I'm not thinking about it in the roadmap yet :)
